src/connection.ts

```diff
@@ -297,6 +297,46 @@ export class Connection extends JSForceConnection {
         });
     });
   }
+
+  /**
+   * Executes a query using either standard REST or Tooling API, returning a single record.
+   * Will throw if either zero records are found OR multiple records are found.
+   *
+   * @param soql The SOQL string.
+   * @param options The query options.
+   */
+  public async singleRecordQuery<T>(
+    soql: string,
+    options: SingleRecordQueryOptions = {
+      choiceField: 'Name',
+    }
+  ): Promise<T> {
+    const result = options.tooling ? await this.tooling.query<T>(soql) : await this.query<T>(soql);
+    if (result.totalSize === 0) {
+      throw new SfdxError(`No records found for ${soql}`, SingleRecordQueryErrors.NoRecords);
+    }
+    if (result.totalSize > 1) {
+      throw new SfdxError(
+        options.returnChoicesOnMultiple && options.choiceField
+          ? `Multiple records found. ${result.records
+              .map((item) => item[(options.choiceField ?? 'Name') as keyof T])
```

Comment:
Given condition of `options.returnChoicesOnMultiple && options.choiceField` it is not possible for `options.choiceField` to be falsey here.

Reply:
yeah, this was from before I put a default on that.  fixing both